pkg/vhdbuilder/datamodel/component_configs.go

```
@@ -124,41 +123,50 @@ func NewKubeProxyImagesFromFile(path string) (*KubeProxyImages, error) {
 	return ret, nil
 }
 
+// processProxyImages is a helper function for ToImageList().
+func processProxyImages(image *ContainerImage, ret *[]string) error {
+	var err error
+	var amd64OnlyImageList []string
+	var multiArchImageList []string
+
+	if image.Amd64OnlyVersions != nil {
+		amd64OnlyImageList, err = toImageList(image.DownloadURL, image.Amd64OnlyVersions)
```

can we return the err immediately if it's not nil here as well?